Tiloaria/assignment3/Table.cpp

```diff
@@ -0,0 +1,113 @@
+#include <cstdio>
+#include <cstdlib>
+#include <iostream>
+#include <set>
+#include <vector>
+#include <string>
+#include <assert.h>
+#include <algorithm>
+
+using namespace std;
+
+/*Dictionary - can find words or his prefics contained inside.*/
+struct Dictionary {
+	set<string> dict, pref;
+	Dictionary (vector<string> new_dict) {
+		for (string cur_el : new_dict) {
+			dict.insert(cur_el);
+			string cur_pref = "";
+			for (int let = 0; let < cur_el.length(); let++) {
+				cur_pref += cur_el[let];
+				pref.insert(cur_pref);
+			}	
+		}
+	}
+	bool isWord(string s) {
+		return (dict.find(s) != dict.end());
+	}
+
+	bool isPrefix(string s) {
+		return (pref.find(s) != pref.end());
+	}
+};
+
+struct Word_matrix;
+
+void rec_find_of_words(set<string> &contains, Word_matrix &cur_mat, Dictionary &dict, vector<vector<bool> > &visited, int wi, int he, string cur_word);
+
+struct Word_matrix {
+	int width = 0, height = 0;
+	vector<vector<char> > grid;
+	Word_matrix (vector<vector<char> > new_grid) {
+		height = new_grid.size();
+		if (height != 0)
+			width = new_grid[0].size();
```

Comment:
What if your grid is:

```
A B
B
A B
```

Consider also checking that the width is the same for the entire grid, not just the first row.